edflow/iterators/template_iterator.py

```diff
@@ -227,3 +228,34 @@ def step_op(self, model, **kwargs):
         written to the log file and stdout. Outputs of 'eval_op' are written
         into the project's eval folder to be evaluated with `edeval`."""
         raise NotImplemented()
+
+    def get_fixed_examples(self, names):
+        """collect fixed examples from dataset with names.
+        
+        Examples
+        --------
+
+        If the dataset returns the examples ["image", "stickman"], (as in VUnet),
+        `self.get_fixed_examples(["image", "stickman"])` will return
+        {
+            "image": batch of self.dset[i]["image"] with fixed i,
+            "stickman": batch of self.dset[i]["stickman"] with fixed i
+        }
+
+        TODO: document usage somewhere
+        TODO: add option to config to automatically run log_op on fixed examples instead of having to do this manually
+        """
+        if not hasattr(self, "fixed_examples"):
+            fixed_random_indices = np.random.RandomState(1).choice(
+                len(self.dataset), self.config["batch_size"]
+            )
+            fixed_example_indices = set_default(
+                self.config, "fixed_example_indices", fixed_random_indices
+            )
+            fixed_examples = {}
+            for n in names:
+                fixed_examples_n = [self.dataset[i][n] for i in fixed_example_indices]
```

Comment:
this is unnecessarily inefficient. Make sure `get_example` is called only once for each index, eg `fixed_examples = [self.dataset[i] for i in fixed_example_indices]` outside of the `for n in names` loop.

Reply:
Done